packages/pigeon/tool/shared/generation.dart

```diff
@@ -0,0 +1,231 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:path/path.dart' as p;
+
+import 'process_utils.dart';
+
+enum GeneratorLanguages {
+  cpp,
+  java,
+  kotlin,
+  objc,
+  swift,
+}
+
+// A map of pigeons/ files to the languages that they can't yet be generated
+// for due to limitations of that generator.
+const Map<String, Set<GeneratorLanguages>> _unsupportedFiles =
+    <String, Set<GeneratorLanguages>>{
+  'enum_args': <GeneratorLanguages>{GeneratorLanguages.cpp},
+};
+
+String _snakeToPascalCase(String snake) {
+  final List<String> parts = snake.split('_');
+  return parts
+      .map((String part) =>
+          part.substring(0, 1).toUpperCase() + part.substring(1))
+      .join();
+}
+
+// Remaps some file names for Java output, since the filename on Java will be
+// the name of the generated top-level class. In some cases this is necessary
+// (e.g., "list", which collides with the Java List class in tests), and in
+// others it is just preserving previous behavior from the earlier Bash version
+// of the generation to minimize churn during the migration.
+// TODO(stuartmorgan): Remove the need for this when addressing
+// https://github.com/flutter/flutter/issues/115168.
+String _javaFilenameForName(String inputName) {
+  const Map<String, String> specialCases = <String, String>{
+    'android_unittests': 'Pigeon',
+    'host2flutter': 'Host2Flutter',
+    'list': 'PigeonList',
+    'message': 'MessagePigeon',
+    'voidflutter': 'VoidFlutter',
+    'voidhost': 'VoidHost',
+  };
+  return specialCases[inputName] ?? _snakeToPascalCase(inputName);
+}
+
+Future<int> generatePigeons({required String baseDir}) async {
```

Comment:
docstrings

Reply:
Thanks, I completely forgot to go back and add them! Done everywhere.